test/test_tagging.py

```diff
@@ -0,0 +1,241 @@
+import json
+from typing import (
+    Optional,
+)
+import unittest
+from unittest.mock import (
+    MagicMock,
+    patch,
+)
+
+from azul.deployment import (
+    populate_tags,
+)
+from azul.logging import (
+    configure_test_logging,
+)
+from azul_test_case import (
+    AzulUnitTestCase,
+)
+
+
+def setupModule():
+    configure_test_logging()
+
+
+class TestTerraformResourceTags(AzulUnitTestCase):
+
+    def permissive_compare(self, expected, actual):
+        """
+        Wrapper for assertDictEqualPermissive doctests.
+
+        >>> t = TestTerraformResourceTags()
+        >>> t.permissive_compare(
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: 789}]},
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: 789}]}
+        ... )
+
+        >>> t.permissive_compare(
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: None}]},
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: 'abc'}]}
+        ... )
+
+        >>> t.permissive_compare(  #doctest: +ELLIPSIS
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: 'def'}]},
+        ...     {'foo': 'bar', 'qaz': [{'qux': 123, 456: 'abc'}]}
+        ... )
+        ...
+        Traceback (most recent call last):
```

Comment:
These traceback literals is what we wanted to avoid in the first place. The doctests should be on the method that does the comparison, not the one that asserts.